tribits/core/package_arch/TribitsWriteClientExportFiles.cmake

```diff
@@ -577,6 +577,20 @@ function(tribits_append_dependent_package_config_file_includes_and_enables packa
       "set(${EXPORT_FILE_VAR_PREFIX}_ENABLE_${depPkg} ${enableVal})\n")
   endforeach()
 
+  # Put in set() statements for exported cache vars
+  string(APPEND configFileStr
+    "\n# Exported cache variables\n")
+  if (NOT "${${packageName}_PARENT_PACKAGE}" STREQUAL "")
+    foreach(exportedCacheVar IN LISTS ${${packageName}_PARENT_PACKAGE}_PKG_VARS_TO_EXPORT)
+      string(APPEND configFileStr
+        "set(${exportedCacheVar} \"${${exportedCacheVar}}\")\n")
```

Comment:
This is a tricky issue.  One could argue that it is better to export the value that an in-project client would see when they evaluated the variable.

How can we put in a check that if a local variable exists in addition to the cache var that they both have the same value?

Reply:
```cmake
if(NOT "${${exportedCacheVar}}" STRQUAL "$CACHE{${exportedCacheVar}}")
  message(WARNING "Cache value of ${exportedCacheVar} does not match local value")
endif()
```

Reply:
The further complicate things, this code gets executed from the package's top-level CMakeLists.txt file scope (through the the call `tribits_package_postprocess()` macro).  So if there is a local variable of the same name then that value will only be used in this one package's scope and not downstream packages.

So to help catch this error, we need to ensure that if a local var exists then it will have the same value.  I will put in an assert and unit test for this.

Reply:
Should be resolved in new commit https://github.com/TriBITSPub/TriBITS/pull/520/commits/903b9a1cd298d87b52550dd6ef420f75e9e3502e.  Can you take a look?

Reply:
Thanks, that looks good to me.